pre-analysis/COMMON_dilution.py

```diff
@@ -0,0 +1,291 @@
+            }
+
+
+    return concentration
+
+
+def create_lines_QIAgility(concentration):
+    """Write one line for each sampleid.
+    Add concentration for each sampleid.
+    """
+
+    logging.info("Creating lines")
+
+    lines = [SEPARATOR["Separator semicolon"].join(HEADER["Header QIAgility"])]
+
+    for sample_name, sample_info in concentration.items():
+        lines.append(f"{sample_info['Sample Name']}{SEPARATOR['Separator semicolon']}{sample_info['Concentration (ng/ul)']:.2f}")
+
+    return lines
+
+
+def create_lines_Beckman(concentration):
+    """Write one line for each well in a 96 well plate. 
+    Add concentration for wells with samples.
+    """
+
+    logging.info("Creating lines")
+
+    lines = [SEPARATOR["Separator tab"].join(HEADER["Header Beckman"])]
+
+    for sample_data, sample_info in concentration.items():
+        lines.append(f"{sample_info['Well']}{SEPARATOR['Separator tab']}{sample_info['Concentration (ng/ul)']:.2f}")
+
+    return lines
+
+
+def create_lines_Manual(concentration):
+    """ Write dilution file. 
+    Add new line for each sampleid.
+    """
+
+    logging.info("Create lines")
+
+    lines = [SEPARATOR["Separator semicolon"].join(HEADER["Header Manual"])]
+
+    for sample_data in concentration.values():
+        sorted_sample_data = [sample_data[header] for header in HEADER["Header Manual"]]
+        line = SEPARATOR["Separator semicolon"].join(str(data) for data in sorted_sample_data)
+        lines.append(line)
+
+    return lines
+
+
+def write_to_file_txt(args, lines):
+    """Write lines to file
+    """
+    filename = args["fileNAME"]+"_dilution-set-up.txt" 
+    logging.info(f"Writing to file {filename}")
+    with open(filename, "w") as dilution_file:
+        dilution_file.write("\n".join(lines))
+
+
+def write_to_file_xlsx(args, lines):
+    """Write lines to file
+    """
+
+    workbook = xlsxwriter.Workbook(args[ "outfile" ] + '.xlsx')
+    worksheet = workbook.add_worksheet()
+
+    logging.info(f"Writing to file")
+
+    cell_format_2dec = workbook.add_format()
+    cell_format_2dec.set_num_format('0.00')
+    cell_format_2dec.set_border()
+    cell_format_2dec.set_border_color('#D9D9D9')
+
+    cell_format_red2dec = workbook.add_format({'bg_color':   '#FFC7CE', 'font_color': '#9C0006'})
+    cell_format_red2dec.set_num_format('0.00')
+    cell_format_red2dec.set_border()
+    cell_format_red2dec.set_border_color('#D9D9D9')
+
+    cell_format_grey = workbook.add_format({'bg_color' : '#F2F2F2', 'font_color' : '#595959'})
+    cell_format_grey.set_border()
+    cell_format_grey.set_border_color('#D9D9D9')
+
+    cell_format_blue = workbook.add_format({'bg_color' : '#a7f0fc'})
+    cell_format_blue.set_border()
+    cell_format_blue.set_border_color('#D9D9D9')
+
+    columns = (15, 15, 15, 15, 25, 25, 25, 18, 18, 18, 18, 40)
+    for i in range(len(columns)):
+        worksheet.set_column(i, i, columns[i])
+
+    xrow = 0
+    lcol = 0
+
+    for i in HEADER["Header Manual"]:
```

Comment:
Same here, add instrument_name as input to the function and get the header using that.
```suggestion
    for i in HEADER[instrument_name]:
```